File: src/main/java/com/menmar/gestionTienda/service/ClienteService.java
package com.menmar.gestionTienda.service;

import com.menmar.gestionTienda.model.PageResponse;
import com.menmar.gestionTienda.model.cliente.ClienteRequest;
import com.menmar.gestionTienda.model.cliente.ClienteResponse;
import org.springframework.data.domain.Pageable;

public interface ClienteService {
    ClienteResponse crear(ClienteRequest request);
    PageResponse<ClienteResponse> listar(Pageable pageable);
    PageResponse<ClienteResponse> buscarPorNombre(String nombre, Pageable pageable);
    ClienteResponse buscarPorId(Long id);
    ClienteResponse buscarPorTelefono(String telefono);
    ClienteResponse actualizar(Long id, ClienteRequest request);
}

File: frontend/src/service/ClienteService.ts
import api from './ApiService'
import type { ClienteRequest, ClienteResponse, PageResponse } from '@/types'

const ClienteService = {
    async listar(nombre?: string, page = 0, size = 20): Promise<PageResponse<ClienteResponse>> {
        const res = await api.get<PageResponse<ClienteResponse>>('/clientes', {
            params: { nombre, page, size }
        })
        return res.data
    },

    async obtener(id: number): Promise<ClienteResponse> {
        const res = await api.get<ClienteResponse>(`/clientes/${id}`)
        return res.data
    },

    async buscarPorTelefono(telefono: string): Promise<ClienteResponse> {
        const res = await api.get<ClienteResponse>(`/clientes/telefono/${telefono}`)
        return res.data
    },

    async crear(req: ClienteRequest): Promise<ClienteResponse> {
        const res = await api.post<ClienteResponse>('/clientes', req)
        return res.data
    },

    async actualizar(id: number, req: ClienteRequest): Promise<ClienteResponse> {
        const res = await api.put<ClienteResponse>(`/clientes/${id}`, req)
        return res.data
    }
}

export default ClienteService
